docs/config/shared-options.md

```diff
@@ -135,7 +136,9 @@ A package with conditional exports may have the following `exports` field in its
 
 Here, `import` and `require` are "conditions". Conditions can be nested and should be specified from most specific to least specific.
 
-Vite has a list of "allowed conditions" and will match the first condition that is in the allowed list. The default allowed conditions are: `import`, `module`, `browser`, `default`, and `production/development` based on current mode. The `resolve.conditions` config option allows specifying additional allowed conditions.
+Some of the default conditions (`production`, `development`, `browser`, `node`) are only applied when the requirements are met. For example, `production` is only applied when `process.env.NODE_ENV === 'production'`, and `browser` is only applied when the environment is `webCompatible`. The `resolve.conditions` config option allows specifying additional allowed conditions and those conditions will be applied unconditionally.
```

Comment:
```suggestion
Some of the default conditions (`production`, `development`, `browser`, `node`) are only applied when the requirements are met. For example, `production` is only applied when `process.env.NODE_ENV === 'production'`, and `browser` is only applied when in the browser environment. The `resolve.conditions` config option allows specifying additional allowed conditions and those conditions will be applied unconditionally.
```
It looks like the PR has removed `webCompatible`

Reply:
```suggestion
Some of the default conditions (`production`, `development`) are only applied when the requirements are met. For example, `production` is only applied when `process.env.NODE_ENV === 'production'`. The `resolve.conditions` config option allows specifying additional allowed conditions and those conditions will be applied unconditionally.
```
Ah, now `browser` and `node` is applied unconditionally, so the sentence should be removed.